airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_table.py

```diff
@@ -171,6 +171,176 @@ def test_given_three_pages_read_then_return_records(self, uuid_mock, mock_auth,
         output = _read(_config(), sync_mode=SyncMode.full_refresh)
         assert len(output.records) == 3
 
+    
+
+    @parameterized.expand(
+            [
+                (["TEST_COLUMN_20"],12, 0,3 , "1970-01-04"),
+                (["TEST_COLUMN_21"], 13,30000, 300001, "1970-01-04T11:20:01.000000"),
+                (["TEST_COLUMN_2"], 11,3, 4, 4),
+                (["TEST_COLUMN_14"], 5,"aa", "ab", "ab")
+            ],
+    )
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_given_cursor_set_state(
+        self,
+        cursor_field,
+        cursor_index,
+        value_first_record,
+        value_second_record,
+        expected_state_cursor_value,
+        uuid_mock, 
+        mock_auth,
+        http_mocker: HttpMocker,
+        ) -> None:
+
+        def snowflake_record():
+            return a_snowflake_response("response_get_table",JsonPath("$.'data'[*]"), cursor_path=JsonPath(f"$.[{cursor_index}]"))
```

Comment:
nit: Should the naming be `a_record`? The naming clash a bit with `snowflake_response` which returns a response while this returns a record